### PR TITLE
new package: dotnet-core-sdk

### PR DESCRIPTION
**Pull request by toffer93**
New package definition for the dotnet-core-sdk from Microsoft

### CODE REVIEW

File: var/spack/repos/builtin/packages/dotnet-core-sdk/package.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    variant('telemetry', default=False, 
+            description='allow collection of telemetry data')
+
+    def setup_environment(self, spack_env, run_env):
+        if '-telemetry' in self.spec:
+            spack_env.set('DOTNET_CLI_TELEMETRY_OPTOUT', 1)
+
+    def install(self, spec, prefix):
+        mkdirp('bin')
+        subprocess.call(['ln', '-rsf', 'dotnet', 'bin/'])
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/dotnet-core-sdk/package.py]: Some of these flags aren't even available on the BSD version of `ln`. You should use `os.symlink`.

File: var/spack/repos/builtin/packages/dotnet-core-sdk/package.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            description='allow collection of telemetry data')
+
+    def setup_environment(self, spack_env, run_env):
+        if '-telemetry' in self.spec:
+            spack_env.set('DOTNET_CLI_TELEMETRY_OPTOUT', 1)
+
+    def install(self, spec, prefix):
+        mkdirp('bin')
+        symlink('../dotnet', 'bin/dotnet')
+        copy_tree(".", join_path(prefix), preserve_symlinks=1)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/dotnet-core-sdk/package.py]: Now that #8289 has been merged, can you replace this with:
```python
install_tree('.', prefix)
```

File: var/spack/repos/builtin/packages/dotnet-core-sdk/package.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    """The .NET Core SDK is a powerful development environment to write
+    applications for all types of infrastructure."""
+
+    homepage = "https://www.microsoft.com/net/"
+    url      = "https://github.com/dotnet/core/"
+
+    version('2.1.300',
+            url='https://download.microsoft.com/download/8/8/5/88544F33-836A'
+                '-49A5-8B67-451C24709A8F/dotnet-sdk-2.1.300-linux-x64.tar.gz',
+            checksum='80a6bfb1db5862804e90f819c1adeebe3d624eae0d6147e5d6694333'
```
[the remark this answers is not in the document]
Comment (tgamblin) [reply var/spack/repos/builtin/packages/dotnet-core-sdk/package.py]: I would prefer that people not use the generic `checksum`, to make packages more readable.  That, however, is not a `sha256`, but a `sha224`.  So either label it `sha224` or just use a `sha256`.

File: var/spack/repos/builtin/packages/dotnet-core-sdk/package.py (hunk 2 above)
Comment (toffer93) [reply var/spack/repos/builtin/packages/dotnet-core-sdk/package.py]: Accidentally I have deleted my folk at github from july (see merge from 'unknow repo'). Therefore I can no longer access this merge request and add new commits to it. Should I open a new merge request or what is the desired action right now? Apologies for this trouble! Thanks

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/dotnet-core-sdk/package.py]: You can open a new pull request and simply delete this one.

**Pull request closed without merging**